sdk/src/main/java/com/mapbox/maps/renderer/egl/EGLCore.kt

```diff
@@ -176,16 +179,37 @@ internal class EGLCore(
     return EGL10.EGL_SUCCESS
   }
 
-  /**
-   * Checks for EGL errors.
-   */
-  private fun checkEglErrorNoException(msg: String): Boolean {
+  internal fun addRendererStateListener(listener: RendererSetupErrorListener) {
+    rendererSetupErrorListenerSet.add(listener)
+  }
+
+  internal fun removeRendererStateListener(listener: RendererSetupErrorListener) {
+    rendererSetupErrorListenerSet.remove(listener)
+  }
+
+  internal fun clearRendererStateListeners() {
+    rendererSetupErrorListenerSet.clear()
+  }
+
+  private fun notifyRendererSetupErrorListenersIfNeeded(functionName: String): Int? {
+    val eglError = checkEglError(functionName)
```

Comment:
btw, if `checkEglError(..)` checks `eglGetError()` and it return `null` as success result, next line will check `null != EGL10.EGL_SUCCESS` and executes block below.

Reply:
I didn't find a way to actually trigger `eglGetError()` return `null`. Even if does - with current logic we will trigger an error `EglError(null)` which is OK as null returned is def some kind of error.

Reply:
testing did actually show functions when `null` is returned and this is not considered to be an error, will fix it. Thanks for pointing it out!

Reply:
in general I did mess up with logic a bit, EGL error could not even be null.